src/awkward/operations/ak_from_buffers.py

```diff
@@ -151,9 +169,7 @@ def _reconstitute(form, length, container, getkey, backend, byteorder, simplify)
     elif isinstance(form, ak.forms.NumpyForm):
         dtype = ak.types.numpytype.primitive_to_dtype(form.primitive)
         raw_array = container[getkey(form, "data")]
-        real_length = length
-        for x in form.inner_shape:
-            real_length *= x
+        real_length = length * math.prod(form.inner_shape)
```

Comment:
I went and checked on when [math.prod](https://docs.python.org/3/library/math.html#math.prod) was added: Python 3.8, and we support Python 3.8 onwards, so this is fine.